tests/prague/eip7702_set_code_tx/test_set_code_txs.py

```diff
@@ -2570,6 +2570,52 @@ def test_set_code_to_precompile(
     )
 
 
+@pytest.mark.with_all_precompiles
+def test_set_code_to_precompile_not_enough_gas_for_precompile_execution(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    precompile: int,
+):
+    """
+    Test set code to precompile and making direct call in same transaction with intrinsic gas
+    only, no extra gas for precompile execution.
+    """
+    auth_signer = pre.fund_eoa(amount=1)
+    auth = AuthorizationTuple(address=Address(precompile), nonce=0, signer=auth_signer)
+
+    intrinsic_gas = fork.transaction_intrinsic_cost_calculator()(
+        authorization_list_or_count=[auth],
+    )
+    discount = min(
+        Spec.PER_EMPTY_ACCOUNT_COST - Spec.PER_AUTH_BASE_COST,
+        intrinsic_gas // 5,  # max discount EIP-3529
+    )
+
+    tx = Transaction(
+        sender=pre.fund_eoa(),
+        to=auth_signer,
+        gas_limit=500_000,
```

Comment:
```suggestion
        gas_limit=intrinsic_gas,
```
Should work the same in theory.